### ISSUE TITLE
Dependency conflicts

### ISSUE THREAD

**Issue by sagotch** [ocsigen/eliom#304]
```
[eliom-base-app] https://github.com/ocsigen/eliom-base-app.git already up-to-date
The following dependencies couldn't be met:
  - eliom-base-app -> eliom >= dev -> tyxml = 3.6.0
  - eliom-base-app -> eliom >= dev -> js_of_ocaml > 2.7
Your request can't be satisfied:
  - js_of_ocaml.dev is in conflict with tyxml.3.6.0
```

For now, I solved it with

```
opam pin add -k git js_of_ocaml https://github.com/ocsigen/js_of_ocaml.git#ed17cef75b472053906bd1e809e0a214c4282fc1
```

**Comment by vasilisp** [comment]
#299 restores compatibility, and I would like to merge it today. There are corresponding `tyxml` branches for `ocsigen-toolkit`, `ocsigen-widgets`, and `eliom-base-app`.

**Comment by vasilisp** [comment]
I merged #299 (and the corresponding `ocsigenserver` branch) to restore compatibility. Pin the preceding commits if that causes problems.

We can merge the `ocsigen-toolkit`, `ocsigen-widgets`, and `eliom-base-app` whenever you want.

**Comment by vasilisp** [comment]
Everything has been merged, so all the master branches are compatible with each other (and with TyXML 4.0).
